### DISCUSSION TITLE
A prototype that helps generate API URL to CL

### DISCUSSION THREAD

**Discussion by legaltextai** [freelawproject/courtlistener#4202] [General]
[CL API URL generator. ](https://helpmefindcase.com/._courtlistener_api_generator)
The idea is to assist with generating correct API URL based on natural language only. E.g, 
"find cases from kansas district courts where judge name is johnson, filed after 2010" 
"show oracle v google cases from federal appellate, and supreme court" 
"782 F. Supp. 2d 1223"
Hope it's helpful.

**Comment by mlissner** [comment]
That's pretty neat. I just tried:

> abortion-related cases relying on roe v wade that aren't in the supreme court

And it did OK. It could have done -court:scotus, but instead it selected the 13 circuit courts:

https://www.courtlistener.com/api/rest/v3/search/?q=abortion+%22roe+v+wade%22&type=o&order_by=score%20desc&stat_Precedential=on&court=ca1%20ca2%20ca3%20ca4%20ca5%20ca6%20ca7%20ca8%20ca9%20ca10%20ca11%20cadc%20cafc

**Comment by legaltextai** [comment]
the tool can improve significantly if you give more tricky examples like this one. 
what should be a correct url for search api for this query?

**Reply by mlissner** [reply]
I don't know. I was curious what it came up with! I think `-court:scotus` would be a better parameter, and I was thinking it'd be neat it if did `cites:(108713 OR 9425157 OR 9425158 OR 9425159)`, but that's asking a lot!

**Reply by legaltextai** [reply]
-court:scotus is not working for me. is it the right filter for search api? 
https://www.courtlistener.com/?q=abortion%20%22roe%20v%20wade%22&type=o&-court:scotus# 
https://www.courtlistener.com/api/rest/v3/search/?q=abortion+%22roe+v+wade%22&type=o&order_by=score%20desc&stat_Precedential=on&-court=scotus  

finding cases that cite certain other cases should be working now 
<img width="822" alt="Screenshot 2024-07-15 at 11 45 57 AM" src="https://github.com/user-attachments/assets/83ed3d65-cf5c-4729-a4a5-4b50765e601b">

**Reply by mlissner** [reply]
Sorry, that was sorta off the cuff. It's supposed to be `-court_id:scotus`

**Comment by sethchandler** [comment]
This is extremely useful. Thank you. It provides a teaching tool where one can learn by example (in addition to providing actually useful queries).  It would also be great if the user could opt to provide the Python requests version of the api call, though ChatGPT and/or Claude is pretty good at translating them anyway. I'm curious how you did this? Fine tune an LLM? Multi-shot learning by providing lots of examples? RAG by feeding in documentation. Some mixture?

**Reply by legaltextai** [reply]
thank you. a mix of those methods, and then update the prompt every time you hit an error with a more complicated query. 

https://github.com/legaltextai/courtlistener_api_generator/blob/main/courtlistener_api_generator.py
